refactor(oxlint): use `vitest`s built in file snapshot comparison

camc314 opened the pull request:
not sure if there was a reason this had a manual impl, but vitest's implementation should handle the line ending proglems (i think)

overlookmotel reviewed (approved):
> not sure if there was a reason this had a manual impl

Reason was that I didn't know this API existed!

camc314 committed:
refactor(oxlint): use `vitest`s built in file snapshot comparison (#14392)

not sure if there was a reason this had a manual impl, but vitest's implementation should handle the line ending proglems (i think)
